### ISSUE TITLE
Review and update OTel semantic conventions

### ISSUE THREAD

**Issue by lmolkova** [Azure/azure-functions-host#10560]
I see a few minor discrepancies:

1. `faas.trigger` is set to some Azure Function constant like `eventHubTrigger`

![Image](https://github.com/user-attachments/assets/baf80c3a-6cf7-4979-8a30-58caead38cdc)

Based on semantic conventions, I believe it should be `pubsub` - https://github.com/open-telemetry/semantic-conventions/blob/main/docs/faas/faas-spans.md and the fact that it's an Event Hub is captured in the corresponding `messaging.system`. 

`faas.trigger` has the following list of well-known values. If one of them applies, then the respective value MUST be used; otherwise, a custom value MAY be used.

| Value  | Description | Stability |
|---|---|---|
| `datasource` | A response to some data source operation such as a database or filesystem read/write | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
| `http` | To provide an answer to an inbound HTTP request | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
| `other` | If none of the others apply | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
| `pubsub` | A function is set to be executed when messages are sent to a messaging system | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
| `timer` | A function is scheduled to be executed regularly | ![Experimental](https://img.shields.io/badge/-experimental-blue) |

2. Very minor: `ai.sdk.prefix = azurefunctions:4.1036.0.0` or `dotnetiso:1.1.0.0` is populated on resources even when I don't use Application Insights SDK and use pure OTel like here https://github.com/lmolkova/testfuncotel. We probably should not set it if AppInsights is not enabled. 

3. `faas.version = 4.1036.0.0` is set on host resources, but it should not be set for Azure functions according to semconv 

   [Azure Functions: Not applicable. Do not set this attribute.](https://github.com/open-telemetry/semantic-conventions/blob/35664be5d835b2f2e8a2e414dea5cef3b04d045a/docs/attributes-registry/faas.md?plain=1#L67)

   Let's fix one side or the other.

4. `faas.execution` is used instead of `faas.invocation_id`. It was right according to OTel schema 1.17 which Functions support, but it was renamed later. Can we update the schema version for OTel case while it's all still experimental?

5. `service.version` on the host is reported as `4.1036.0.0` which does not seem right - it seems to be internal Function host version, but I assume users would expect it to be something about their code (if we can infer it). 
   It maybe just something to document - the 4.1036.0.0 is probably automatically picked up by otel-dotnet from the assembly version and is probably easily overridden with OTEL env var.

Happy to discuss further and help if you'd like to change anything in the otel semantic conventions.

**Comment by stebet** [comment]
@RohitRanjanMS It would be awesome if the function host set the root span name to the  `faas.name` or the HTTP method + `http.route` values for incoming HTTP requests. Currently the root spans are just named after the request method (`GET`, `POST`, etc.).

Example:

<img width="1274" alt="Image" src="https://github.com/user-attachments/assets/38573cf4-90c4-45f6-9ba2-e178d0974f64" />

Would that make sense @lmolkova?

**Comment by lmolkova** [comment]
yep, I think it's not the case because `http.route` is not available for some reason. I wonder if we can force-update this activity name to `{verb} {function name}`.

Tried to check what FaaS semconv recommend and I think they are not clear enough - https://github.com/open-telemetry/semantic-conventions/issues/1858

**Comment by stebet** [comment]
The trigger route is available from the `Raw` property on the `context.FunctionMetadata` here: https://github.com/Azure/azure-functions-host/blob/cfa63f6c4e149363fc3792608b6c3581e3501fda/src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs#L1724

Here's an example where the `route` property is fetched:
https://github.com/Azure/azure-functions-host/blob/cfa63f6c4e149363fc3792608b6c3581e3501fda/src/WebJobs.Script.WebHost/Extensions/FunctionMetadataExtensions.cs#L227

I can try making a PR, I have no idea how easy or hard it is to run the host locally but let's see :)

**Comment by RohitRanjanMS** [comment]
This is done. Will be available in host 1039 release.

**Comment by mattchenderson** [comment]
@RohitRanjanMS could you please link any PRs that addressed this?

**Comment by RohitRanjanMS** [comment]
Here's the PR - https://github.com/Azure/azure-functions-host/pull/10929
